libraries/Microsoft.Bot.Builder.Azure.Blobs/BlobsTranscriptStore.cs

```diff
@@ -0,0 +1,419 @@
+                                if (blobItem.Name == continuationToken)
+                                {
+                                    // we found continuation token
+                                    continuationToken = null;
+                                }
+                            }
+                            else
+                            {
+                                blobs.Add(blobItem);
+                                if (blobs.Count == PageSize)
+                                {
+                                    break;
+                                }
+                            }
+                        }
+                    }
+
+                    // Get the continuation token and loop until it is empty.
+                    token = blobPage.ContinuationToken;
+                }
+            }
+            while (!string.IsNullOrEmpty(token) && blobs.Count < PageSize);
+
+            pagedResult.Items = blobs
+                .Select(async bl =>
+                {
+                    var blobClient = _containerClient.Value.GetBlobClient(bl.Name);
+                    return await GetActivityFromBlobClientAsync(blobClient).ConfigureAwait(false);
+                })
+                .Select(t => t.Result)
+                .ToArray();
+
+            if (pagedResult.Items.Length == PageSize)
+            {
+                pagedResult.ContinuationToken = blobs.Last().Name;
+            }
+
+            return pagedResult;
+        }
+
+        /// <summary>
+        /// List conversations in the channelId.
+        /// </summary>
+        /// <param name="channelId">Channel Id.</param>
+        /// <param name="continuationToken">Continuatuation token to page through results.</param>
+        /// <returns>A <see cref="Task"/> A task that represents the work queued to execute.</returns>
+        public async Task<PagedResult<TranscriptInfo>> ListTranscriptsAsync(string channelId, string continuationToken = null)
+        {
+            const int PageSize = 20;
+
+            if (string.IsNullOrEmpty(channelId))
+            {
+                throw new ArgumentNullException($"missing {nameof(channelId)}");
+            }
+
+            string token;
+            List<TranscriptInfo> conversations = new List<TranscriptInfo>();
+            do
+            {
+                token = null;
+
+                var resultSegment = _containerClient.Value.GetBlobsAsync(BlobTraits.Metadata, prefix: $"{SanitizeKey(channelId)}/")
+                                    .AsPages(token);
+
+                await foreach (var blobPage in resultSegment)
+                {
+                    foreach (BlobItem blobItem in blobPage.Values)
+                    {
+                        // Unescape the Id we escaped when we saved it
+                        var conversation = new TranscriptInfo() { Id = Uri.UnescapeDataString(blobItem.Name.Split('/').Where(s => s.Length > 0).Last()), ChannelId = channelId };
```

Comment:
Use Last(s=>s.Length...) instead of Where.

Reply:
nice catch!